### PR TITLE
Rename special to standalone

### PR DESCRIPTION
**Pull request by carmocca**
## What does this PR do?

Special was a confusing name. This PR uses standalone which represents better what it does.

Other candidates were:

`single_process=True`
`split_process=True`
`separate_process=True`

### Does your PR introduce any breaking changes? If yes, please list them.

None

## Before submitting

- [ ] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [x] Did you make sure to **update the documentation** with your changes? (if necessary)
- [n/a] Did you write any **new necessary tests**? (not for typos and docs)
- [n/a] Did you verify new and **existing tests pass** locally with your changes?
- [x] Did you list all the **breaking changes** introduced by this pull request?
- [n/a] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

cc @carmocca @akihironitta @borda @justusschock @awaelchli

### CODE REVIEW

**Review by SeanNaren — APPROVED** [review]

File: tests/utilities/test_warnings.py [hunk 1]
```diff
@@ -21,7 +21,7 @@
 
 from pytorch_lightning.utilities.warnings import _warn, rank_zero_deprecation, rank_zero_warn, WarningCache
 
-running_special = os.getenv("PL_RUNNING_SPECIAL_TESTS", "0") == "1"
+running_special = os.getenv("PL_RUN_STANDALONE_TESTS", "0") == "1"
 if running_special:
```
Comment (rohitgr7) [inline tests/utilities/test_warnings.py:25]:
```suggestion
running_standalone = os.getenv("PL_RUN_STANDALONE_TESTS", "0") == "1"
if running_standalone:
```

File: tests/trainer/test_trainer.py [hunk 2]
```diff
@@ -1462,7 +1462,7 @@ def test_trainer_predict_cpu(tmpdir, datamodule, enable_progress_bar):
     predict(tmpdir, datamodule=datamodule, enable_progress_bar=enable_progress_bar)
 
 
-@RunIf(min_gpus=2, special=True)
+@RunIf(min_gpus=2, standalone=True)
```
Comment (rohitgr7) [inline tests/trainer/test_trainer.py:1465]: test name needs update.

File: tests/special_tests.sh [hunk 3]
```diff
@@ -15,12 +15,12 @@
 set -e
 
 # this environment variable allows special tests to run
-export PL_RUNNING_SPECIAL_TESTS=1
+export PL_RUN_STANDALONE_TESTS=1
```
Comment (rohitgr7) [inline tests/special_tests.sh:18]: maybe update the filename too?

**Review by rohitgr7 — APPROVED** [review]
small comments!

**Review by justusschock — APPROVED** [review]

**Review by Borda — APPROVED** [review]

**Pull request merged**